packages/terra-worklist-data-grid/src/WorklistDataGrid.jsx

```diff
@@ -357,37 +357,46 @@ function WorklistDataGrid(props) {
     }
   };
 
-  const selectMultipleRows = useCallback((newEndRowIndex, rangeInfo) => {
-    if (!onRowSelect || !rangeInfo) {
-      return;
+  const getRowsInRangeToUnselect = useCallback((startOfRange, prevEndOfRange, newEndOfRange) => {
+    let rowIdsToUnselect = [];
+    if (!prevEndOfRange) {
+      // short-circuit since there was no previous range, there is nothing to unselect.
+      return rowIdsToUnselect;
+    }
+
+    if (startOfRange > prevEndOfRange) {
+      // The range extends upward from the anchor row
+      if (newEndOfRange > prevEndOfRange) {
+        // The range was moved down towards the anchor so rows that no longer qualify for the range need to be unselected.
+        rowIdsToUnselect = rows.slice(prevEndOfRange - 1, Math.min(newEndOfRange, startOfRange)).map(row => row.id);
+      }
+    } else if (startOfRange < prevEndOfRange) {
+      // The range extends downward from the anchor row
+      if (newEndOfRange < prevEndOfRange) {
+        // The range was moved up towards the anchor so rows that no longer qualify for the range need to be unselected. New endRangeIndex becomes ordered End
+        rowIdsToUnselect = rows.slice(Math.max(startOfRange, newEndOfRange), prevEndOfRange).map(row => row.id);
+      }
     }
+    return rowIdsToUnselect;
+  }, [rows]);
 
+  const selectMultipleRows = useCallback((startOfRange, prevEndOfRange, newEndOfRange) => {
     if (!hasSelectableRows) {
       onEnableRowSelection();
     }
+    if (!onRowSelect) {
+      return;
+    }
```

Comment:
If the consumer does not have an onRowSelect callback, do we need to enable row selection?

Reply:
We think so since row selection can still be enabled from outside of the grid with/without onRowSelect.